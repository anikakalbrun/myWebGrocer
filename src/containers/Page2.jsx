import React, {Component} from 'react';
import { Field, reduxForm } from 'redux-form';
import {func, bool} from 'prop-types';

import Button from '../../src/components/Button.jsx';
import FormField from '../components/FormField';
import DateField from '../components/DateField';
import Wrapper from '../components/Wrapper';
import post from '../utils/post'
import {required, alf, maxLength30, maxLength60, minLength2, alf_space_dash, email} from '../utils/formValidation';
import {setPageNumber} from '../actions'
import '../style/css/Page.css'

export class Page2 extends Component {

    static propTypes = {
        submitting: bool.isRequired,
        invalid: bool.isRequired,
        handleSubmit: func.isRequired
    };

    render() {
        const {handleSubmit, pristine, submitting} = this.props;
        return (
            <Wrapper innerContainerClass='inverse default large'>
                <div className='row'>
                    <form className="form-horizontal" onSubmit={handleSubmit}>
                        <div className='col-xs-12 col-md-6'>
                            <Field name="firstName"
                                   component={FormField}
                                   label="First Name"
                                   placeholder="John"
                                   autoComplete='given-name'
                                   validate={[required, alf, minLength2, maxLength60]}/>

                            <Field name="dob"
                                   component={DateField}
                                   label="Date of Birth"
                                   placeholder="30/01/1975"
                                   showYearDropdown = {true}
                                   validate={[required]}/>

                            <Field name="email"
                                   component={FormField}
                                   label="Email Address"
                                   placeholder="john.doe@gmail.com"
                                   type='email'
                                   autoComplete='email'
                                   validate={[required, email]}/>
                        </div>

                        <div className='col-xs-12 col-md-6'>
                            <Field name="surname"
                                   component={FormField}
                                   label="Surname"
                                   placeholder="Doe"
                                   autoComplete='family-name'
                                   validate={[required, alf_space_dash, minLength2, maxLength60]}/>

                            <Field name="nationality"
                                   component={FormField}
                                   label="Nationality"
                                   placeholder="Irish"
                                   autoComplete='country-name'
                                   validate={[required, alf, minLength2, maxLength30]}/>

                            <Field name="occupation"
                                   component={FormField}
                                   label="Occupation"
                                   placeholder="Front End Developer"
                                   validate={[required, alf_space_dash, minLength2, maxLength60]}/>
                        </div>

                        <Button type='submit'
                                disabled={pristine || submitting}
                                text='Submit'
                                btnClassName='button--green pull-right m-r-15'/>
                    </form>
                </div>
            </Wrapper>
        );
    }
}

Page2 = reduxForm({
    form: 'RegistrationForm',
    destroyOnUnmount: true,
    onSubmit: post,
    url: '/users',
    callback: setPageNumber,
    pageNumber: 3
})(Page2);

export default Page2;
