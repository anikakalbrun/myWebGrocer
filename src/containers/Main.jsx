import React, {Component} from 'react';
import {connect} from "react-redux";
import {number, func} from 'prop-types';

import Page1 from '../components/Page1';
import Page2 from '../containers/Page2';
import Page3 from '../components/Page3';
import {setPageNumber} from '../actions';
import '../style/css/Main.css';

const getHeight = (innerHeight) => innerHeight - 91;

export class Main extends Component {

    static propTypes = {
        pageNumber: number.isRequired,
        setPageNumber: func.isRequired
    };

    state = {
        pageNumber: getHeight(window.innerHeight)
    };

    componentDidMount() {
        window.addEventListener("resize", this.setComponentHeight);
        this.setComponentHeight();
    };

    componentWillUnmount() {
        window.removeEventListener("resize", this.setComponentHeight);
    }

    setComponentHeight = () => {
        this.setState({minHeight: getHeight(window.innerHeight)})
    };

    goToSecondPage = () => {
        this.props.setPageNumber(2)
    };

    render() {
        const {pageNumber} = this.props;
        return (
            <main className='main'>
                <div style={{minHeight: this.state.minHeight, height: 'auto', overflow: 'auto'}}>
                    {pageNumber === 1 && <Page1 goToSecondPage={this.goToSecondPage}/>}
                    {pageNumber === 2 && <Page2/>}
                    {pageNumber === 3 && <Page3/>}
                </div>
            </main>
        );
    }
}

function mapStateToProps(state) {
    return {
        pageNumber: state.pageReducer.pageNumber
    }
}

export default connect(mapStateToProps, {setPageNumber})(Main)