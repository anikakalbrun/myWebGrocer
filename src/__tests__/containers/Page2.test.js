import React from 'react';
import configureMockStore from 'redux-mock-store';
import {Provider} from "react-redux";

import { Page2 } from '../../containers/Page2';

let wrapper;
const mockStore = configureMockStore();
const handleSubmit = jest.fn();
const props = {
    submitting : false,
    touched : false,
    error : null,
    handleSubmit,
};

describe('Page2', () => {

    it('renders correctly', () => {
        wrapper = renderer.create(<Provider store={mockStore({})}><Page2 {...props}/></Provider>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    beforeEach(() => {
        wrapper = mount(<Provider store={mockStore({})}><Page2 {...props}/></Provider>);
    });

    it('has an input with attribute [name="firstName"]', () => {
        expect(wrapper.find('input[name="firstName"]').length).toEqual(1);
    });

    it('has an input with attribute [name="surname"]', () => {
        expect(wrapper.find('input[name="surname"]').length).toEqual(1);
    });

    it('has an input with attribute [name="email"]', () => {
        expect(wrapper.find('input[name="email"]').length).toEqual(1);
    });

    it('has an day input', () => {
        expect(wrapper.find('.react-datepicker__input-container > input').length).toEqual(1);
    });

    it('has an input with attribute [name="nationality"]', () => {
        expect(wrapper.find('input[name="nationality"]').length).toEqual(1);
    });

    it('has an input with attribute [name="occupation"]', () => {
        expect(wrapper.find('input[name="occupation"]').length).toEqual(1);
    });

    it('has an button type submit', () => {
        expect(wrapper.find('button[type="submit"]').length).toEqual(1);
    });
});
