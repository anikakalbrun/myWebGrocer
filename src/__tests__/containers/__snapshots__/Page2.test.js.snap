// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page2 renders correctly 1`] = `
<div
  className="hero__vertical-container"
>
  <div
    className="hero__title inverse default large"
  >
    <div
      className="row"
    >
      <form
        className="form-horizontal"
        onSubmit={[Function]}
      >
        <div
          className="col-xs-12 col-md-6"
        >
          <div
            className="form-group"
          >
            <label
              className="col-xs-12"
            >
              First Name
              :
            </label>
            <div
              className="col-xs-12"
            >
              <input
                autoComplete="given-name"
                className="form-control "
                disabled={false}
                name="firstName"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="John"
                type="text"
                value=""
              />
              <div
                className="no-error"
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-xs-12"
            >
              Date of Birth
              :
            </label>
            <div
              className="col-xs-12"
            >
              <div>
                <div
                  className="react-datepicker-wrapper"
                >
                  <div
                    className="react-datepicker__input-container"
                  >
                    <input
                      autoComplete={undefined}
                      autoFocus={undefined}
                      className="form-control "
                      disabled={false}
                      id={undefined}
                      name={undefined}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="30/01/1975"
                      readOnly={undefined}
                      required={undefined}
                      tabIndex={undefined}
                      title={undefined}
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                className="no-error"
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-xs-12"
            >
              Email Address
              :
            </label>
            <div
              className="col-xs-12"
            >
              <input
                autoComplete="email"
                className="form-control "
                disabled={false}
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="john.doe@gmail.com"
                type="email"
                value=""
              />
              <div
                className="no-error"
              />
            </div>
          </div>
        </div>
        <div
          className="col-xs-12 col-md-6"
        >
          <div
            className="form-group"
          >
            <label
              className="col-xs-12"
            >
              Surname
              :
            </label>
            <div
              className="col-xs-12"
            >
              <input
                autoComplete="family-name"
                className="form-control "
                disabled={false}
                name="surname"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Doe"
                type="text"
                value=""
              />
              <div
                className="no-error"
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-xs-12"
            >
              Nationality
              :
            </label>
            <div
              className="col-xs-12"
            >
              <input
                autoComplete="country-name"
                className="form-control "
                disabled={false}
                name="nationality"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Irish"
                type="text"
                value=""
              />
              <div
                className="no-error"
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-xs-12"
            >
              Occupation
              :
            </label>
            <div
              className="col-xs-12"
            >
              <input
                autoComplete="off"
                className="form-control "
                disabled={false}
                name="occupation"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Front End Developer"
                type="text"
                value=""
              />
              <div
                className="no-error"
              />
            </div>
          </div>
        </div>
        <button
          className="btn button--green pull-right m-r-15"
          disabled={true}
          onClick={undefined}
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
